lib/matplotlib/axes/_axes.py

```diff
@@ -6476,22 +6476,20 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
             range of x.
 
         density : bool, default: False
-            If ``True``, the first element of the return tuple will
-            be the raw counts per bin normalized to form a probability density
-            so that the area under the histogram will integrate to 1, i.e.
-            ``np.sum(n * np.diff(bins)) == 1``.  (Computed by dividing the
-            raw counts ``n0`` by the total number of data points and
-            multiplying by the bin width: ``n = n0 / sum(n0) * np.diff(bins)``)
+            If ``True``, draw and return a probability density: each bin
+            will display the bin's raw count *divided by the bin width*
+            (``n = n0 / sum(n0) * np.diff(bins)``), so that the area under
+            the histogram integrates to 1 (``np.sum(n * np.diff(bins)) == 1``).
```

Comment:
I think there is a mistake that slipped in via https://github.com/matplotlib/matplotlib/pull/16149 (where the text was changed after the first approving review). 

So initially it said *"This is achieved by dividing the count by the number of observations times the bin width [..]"*, which is correct, but ambiguous. But that turned into  `n = n0 / sum(n0) * np.diff(bins)`, which is incorrect and should rather be `n = n0 / (sum(n0) * np.diff(bins))`


I also find `n0` used here adds to confusion. So `n` already has two meanings in this docstring, namely the histogram counts, which are returned, as well as the dimensionality of the input data. `n0` is then then histogram counts in case of multiple data input, but also the "raw counts". 
Hence ideally the complete docstring is revised to use one letter for the return, and another for the dimensionality.

Reply:
Fair point.  I just renamed them "density" and "counts" :)